src/Microsoft.OData.Client/DataServiceClientFormat.cs

```diff
@@ -362,10 +362,10 @@ private static string ChooseMediaType(bool hasSelectQueryOption)
         {
             if (hasSelectQueryOption)
             {
-                return MimeApplicationJsonODataLightWithAllMetadata;
+                return MimeApplicationJsonODataLight;
             }
 
-            return MimeApplicationJsonODataLight;
+            return MimeApplicationJsonODataLightWithAllMetadata;

```

Comment:
Could this cause breaking changes?

Reply:
I'm not really sure. At least it fixes my issue and didn't break any other tests. I tried to find something in the standard about the requirements for the header, but didn't find something. Since there was only one reproducable case [here](https://github.com/OData/odata.net/issues/2567#issuecomment-1384168044) it's more on the server-side where this can cause problems, and my Navision server is having a problem when the header is set to "metadata.full" on a "select-query".
As I interprete this function, the 2 values just got mixed up. I tried to clarify this question [here](https://github.com/OData/odata.net/issues/2567#issuecomment-1403638600) but didn't get an answer.

Reply:
Based on the doc comment:
```c#
 /// <param name="hasSelectQueryOption">
 ///   Whether or not the select query option is present in the request URI.
 ///   If true, indicates that the client should ask the server to include all metadata in a JSON-Light response.
 /// </param>
```
It seems that it was intentional to return all metadata when select query option is enabled, and not just a matter of getting the two values mixed up. 

I don't have the full context of why this implemented this way, I have tried to trace its history using `git blame` but it looks like it may predate the git repo.

If setting metadata to full by default for select queries has issues, then it might be worthwhile to default to minimal metadata.

But I don't think it's a good idea to switch the default to full metadata for all other scenarios like what you are PR is doing. If other scenarios work well so far, I think we should keep using minimal metadata for them.

In conclusion, maybe we should default to minimal metadata for all cases instead of switching (both with and without select)